extension/src/Runner/testRunner.ts

```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { ITestRunnerContext } from "./testRunnerContext";
+
+export interface ITestRunner {

```

Comment:
I don't see it necessary to use `interface` instead of `class`. In JavaScript, there is no built-in `interface`. It is introduced by TypeScript to enforce type checking. The type checking is ideal for check object shape, e.g. checking the object inside a remote call response.
But in our case, we only need polymorphism. Classes work perfectly for that purpose. Using interfaces is an overkill in my opinion.

Reply:
because i have no default behavior for the run method. So it is abstract class versus interface, from my point, class is also a syntax sugar in js, why not interface?